pkg/logging/logging.go

```diff
@@ -117,20 +166,22 @@ func SetupLogging(loggers []string, logOpts map[string]string, tag string, debug
 // SetLogLevel sets the log level on DefaultLogger. This logger is, by
 // convention, the base logger for package specific ones thus setting the level
 // here impacts the default logging behaviour.
-// This function is thread-safe when logging, reading DefaultLogger.Level is
+// This function is thread-safe when logging, reading DefaultLogger.LevelOpt is
 // not protected this way, however.
 func SetLogLevel(level logrus.Level) {
 	DefaultLogger.SetLevel(level)
 }
 
-// ToggleDebugLogs switches on or off debugging logs. It will select
-// DefaultLogLevel when turning debug off.
+// ConfigureLogLevel configures the logging level of the global logger. If
+// debugging is not enabled, it will set the logging level based off of the
+// logging options configured at bootstrap. Debug being enabled takes precedence
+// over the configuration in the logging options.
 // It is thread-safe.
-func ToggleDebugLogs(debug bool) {
+func ConfigureLogLevel(debug bool) {
```

Comment:
When I saw the name change I was expecting this function to take in both `debug` and `options`. It feels weird to mix using the global and the `debug` parameter. Is there any chance we can pass in the configuration here every time? That will probably mean exposing `options` in some way but you might have done all this to avoid that?

Reply:
I can't import `pkg/option` into here because it will result in a dependency loop. Perhaps I can do it as an interface.

Reply:
Oh, I meant that we were referring to the global variable `options` in this package while also taking the `debug` parameter. Both are configuration to be parsed/used. To be clear, I'm fine with the global variable. I would just prefer to pass in both `debug` and `options` here.
I don't fully see why pkg/options would come up in this case; all calls to `ConfigureLogLevel` could pass in the `LogOptions` type (`SetupLogging already does this`).

Reply:
The weird thing about `debug` is that it's extracted from other configuration outside of the logging package. What I'd love for us to have (which would be a significant change) is to be able to register objects to be updated when configuration that could be changed after Cilium is started (e.g., `cilium config x=y`), and that way we'd not have to pass `debug` around as a variable in function calls (not sure how much that makes sense). 

Logging level is an option configured at agent bootstrap vs. debug being a configurable option that can change during a single run of a given agent, which is why debug is the only "variable" here.